### ISSUE TITLE
Is there a reason that `patch` is restricted to `Equatable` elements?

### ISSUE THREAD

**Issue by chrisco314** [tonyarnold/Differ#17]
This feels like a cut and paste issue, but `patch` seems unnecessarily constrained to `Equatable` when there is an equality closure on `diff`, and it compiles without the `Equatable` requirement, see the example below.  

Am I missing something?

```
public extension Diff {

    /// Generates a patch sequence based on a diff. It is a list of steps to be applied to obtain the `to` collection from the `from` one.
    ///
    /// - Complexity: O(N)
    ///
    /// - Parameters:
    ///   - from: The source collection (usually the source collecetion of the callee)
    ///   - to: The target collection (usually the target collecetion of the callee)
    /// - Returns: A sequence of steps to obtain `to` collection from the `from` one.
    public func patch<T: Collection>(
        from: T,
        to: T
    ) -> [Patch<T.Iterator.Element>] where T.Iterator.Element: Equatable {
        var shift = 0
        return map { element in
            switch element {
            case let .delete(at):
                shift -= 1
                return .deletion(index: at + shift + 1)
            case let .insert(at):
                shift += 1
                return .insertion(index: at, element: to.itemOnStartIndex(advancedBy: at))
            }
        }
    }

    /// Generates a patch sequence based on a diff. It is a list of steps to be applied to obtain the `to` collection from the `from` one.
    ///
    /// - Complexity: O(N)
    ///
    /// - Parameters:
    ///   - from: The source collection (usually the source collecetion of the callee)
    ///   - to: The target collection (usually the target collecetion of the callee)
    /// - Returns: A sequence of steps to obtain `to` collection from the `from` one.
    public func patchDoesntNeedEquatable<T: Collection>(
        from: T,
        to: T
        ) -> [Patch<T.Iterator.Element>] {
        var shift = 0
        return map { element in
            switch element {
            case let .delete(at):
                shift -= 1
                return .deletion(index: at + shift + 1)
            case let .insert(at):
                shift += 1
                return .insertion(index: at, element: to.itemOnStartIndex(advancedBy: at))
            }
        }
    }
}
```

Thanks.

**Comment by tonyarnold** [comment]
Honestly, I'm not really sure, I haven't looked at that too closely yet! I'll happily accept PRs if you think the restriction is unnecessary.

**Comment by wokalski** [comment]
It's redundant I think.

**Comment by tonyarnold** [comment]
Fixed by #25 - thanks @garnett
